### PR TITLE
Successful ROCm pytorch build without hcrng and hcsparse.

### PR DESCRIPTION
**Pull request by jithunnair-amd**
Commented out any references to hcrng and hcsparse. Also commented out references to std::max/std::log/std::exp functions in Embedding.cu and SoftMax.cu to get rid of linking errors.

### CODE REVIEW

File: aten/src/ATen/native/cuda/Embedding.cu [hunk 1]
```diff
@@ -180,15 +180,15 @@ __global__ void renorm_kernel(
     } else if (norm_type == 2) {
       v += x * x;
     } else {
-      v += std::pow(x, norm_type);
+      //v += std::pow(x, norm_type);
```
Comment (whchung) [inline aten/src/ATen/native/cuda/Embedding.cu]: @jithunnair-amd I don't think it's a good way to change the behavior of an operator. I'd be in favor of disabling the whole operator instead of changing its behavior just to get things compiled. It would trigger future issue later on. Moreover not comment has been provided on these changes which makes it very hard to fix them later on.

File: aten/src/ATen/native/cuda/Gesv.cu [hunk 2]
```diff
@@ -46,17 +46,17 @@ void magmaGesvBatched<double>(
...
-  magma_queue_t magma_queue;
-  magma_queue_create_from_cuda(
-      tensor.get_device(),
-      context.getCurrentCUDAStream(),
-      THCState_getCurrentBlasHandle(context.getTHCState()),
-      THCState_getCurrentSparseHandle(context.getTHCState()),
-      &magma_queue);
-  return magma_queue;
-}
+//static magma_queue_t createMagmaQueue(const Tensor& tensor) {
```
Comment (whchung) [inline aten/src/ATen/native/cuda/Gesv.cu]: @jithunnair-amd same as the rest of the PR. No comment has been made regarding to the rationale disabling the logic crucial to the operator here. Instead of disabling the implementation of a particular operator I'd like to have the operators themselves be disabled.

**Comment by jithunnair-amd** [comment]
I'll update the changes to be ifdef'd out only for ROCm builds and add appropriate comments, and submit a new PR.
